framework/merchant_api/domain/entities/coles_product_offer.py

```diff
@@ -0,0 +1,19 @@
+from typing import List, Optional
+from pydantic import BaseModel
+
+
+class ColesProductOffer(BaseModel, extra='allow'):
+    class ImageInfo(BaseModel, extra='allow'):
+        uri: str
+
+    class Pricing(BaseModel, extra='allow'):
+        now: float
```

Comment:
current_price & previous_price? Not sure that actually improves it, resolve if not.

Reply:
can't actually change this

when the data comes through from coles/woolies, that is then turned into these models with the .model_validate() method from pydantic

the way that works is by matching key/value pairs to the names of attributes on the class, so therefore the classes must mirror the data structure that comes through exactly